### PR TITLE
Check in/67417/add api calls for new pre check in flow

### PR DESCRIPTION
**Pull request by wullaski**
## Summary

- Creates a hook from useEffect in pre check in complete page
- uses the hook on new unified appointment list if demographics are up to date

## Related issue(s)

- [department-of-veterans-affairs/va.gov-team#67417](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67417)

## Testing done

- unit

## What areas of the site does it impact?

PCI -> Pre-check-in 

## Acceptance criteria

- [ ] It calls the endpoint to set pre check in to complete if demographics are up to date on the complete page and the new appointments page
- [ ] unit tests for new components and hook

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### CODE REVIEW

File: src/applications/check-in/components/ActionItemDisplay.jsx [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
-  const { t } = useTranslation();
 
   const displaySuccessAlert =
-    app === APP_NAMES.PRE_CHECK_IN && preCheckinAlreadyCompleted(appointments);
-
-  const successMessage = hasPhoneAppointments(appointments)
-    ? t('your-provider-will-call-you-at-your-appointment-time')
-    : t('you-can-check-in-with-your-smartphone');
+    app === APP_NAMES.PRE_CHECK_IN &&
+    (preCheckinAlreadyCompleted(appointments) || pages.length < 5);
```
Comment (brianseek) [inline src/applications/check-in/components/ActionItemDisplay.jsx:29]: its unlikely but possible for any one of the demos needing update on it own. You may want to write a little util that checks to make sure non of the demo pages are in the form to determine this.

File: src/applications/check-in/locales/en/translation.json [hunk 2]
```diff
@@ -279,5 +279,6 @@
   "phone-appointments": "Phone appointments",
   "check-in-now-for-your-date-time-appointment": "Check in now for appointment on {{ date, day }}, {{ date, monthDay }} at {{ date, time }}",
   "details-for-appointment-on-date-at-time": "Details for {{ type }} appointment on {{ dateTime, date }} at {{ dateTime, time }}",
-  "appointment-on-date-at-time": "Appointment on {{dateTime, date}} at {{ dateTime, time}}"
+  "appointment-on-date-at-time": "Appointment on {{dateTime, date}} at {{ dateTime, time}}",
+  "setting-pre-check-in-completed": "Setting Pre-Check-In completed"
```
Comment (brianseek) [inline src/applications/check-in/locales/en/translation.json:283]: run this content by UX before finishing this up

**Comment by brianseek** [comment]
I think this approach is looking good for sure!

File: src/applications/check-in/components/ActionItemDisplay.jsx (hunk 1 above)
Comment (wullaski) [reply src/applications/check-in/components/ActionItemDisplay.jsx:29]: We can spin this off in another ticket

File: src/applications/check-in/components/PreCheckInSuccessAlert.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        message={t('setting-pre-check-in-completed')}
+      />
+    );
+  }
+
+  return (
+    <section data-testid="pre-check-in-success-alert">
+      <va-alert
+        close-btn-aria-label="Close notification"
+        closeable
```
Comment (brianseek) [inline src/applications/check-in/components/PreCheckInSuccessAlert.jsx:35]: not related to this ticket, but I noticed that these alerts don't close. I wonder if we need to pass a close func to it?

**Review by brianseek — APPROVED** [review]
:shipit: Looks great!

**Pull request merged**